symengine/complex.h

```diff
@@ -17,6 +17,7 @@ class ComplexBase : public Number
 public:
     virtual RCP<const Number> real_part() const = 0;
     virtual RCP<const Number> imaginary_part() const = 0;
+    virtual RCP<const Number> conjugate() const = 0;
```

Comment:
I think `conjugate` should be a virtual function of Number, so that classes that can be either real or imaginary can override it.